### ISSUE TITLE
Gopher protocol links are highlighted but unclickable

### ISSUE THREAD

**Issue by htrob** [tootsuite/mastodon#7994]
This is more a question than a bug report, because there may be a good reason they've been made this way, I'm just unable to see the reason for the inconsistency. Why they are recognised as links, and marked up using an <a> anchor tag, but they're made unclickable?
![screenshot at 2018-07-11 09-21-59](https://user-images.githubusercontent.com/23458/42571142-93a52dc0-84ec-11e8-9f89-7d23a2c149d4.png)

See attached screenshot.

**Comment by lfuelling** [comment]
Regular links also have those attributes.

You need a handler for the `gopher://` protocol installed for link clicks to work. (And as far as I remember you need to confirm the app that opens this protocol at least in chrome.)

**Comment by nightpool** [comment]
possibly a bug that was introduced when we hacked in gopher support in #7810

**Comment by Gargron** [comment]
Right. People asked to whitelist those protocols, and I did, but seems like the Twitter URL extractor code is incapable of actually retrieving the href here somehow..?

**Comment by algernon** [comment]
[We've tried to look](https://mastodon.social/@ckeen/100383173167807192) a bit deeper into the issue, and one interesting thing I found is that when I open the [original toot](https://octodon.social/@kensanata/100382998406506419), the link is fine, points to the right place. But when I search the toot on my own instance (running Mastodon 2.4.3), or on mastodon.social, we see the same problem as in the screenshot above: the link has no `href`, and the `title` attribute is `null`. It's not a frontend issue either, because I get the same data through the API.

In short, the URL extractor does work, but the link - and the toot - gets butchered somewhere during federation, by the looks of it.

**Comment by algernon** [comment]
FWIW, same problem can be seen when using Pinafore, browser doesn't matter either. This - I believe - strengthens the suspicion that the corruption happens between the origin server and the rest.

I also registered on octodon.social, and searching for the original toot URL there yielded working links, both when using the mastodon frontend, and with Pinafore.

**Comment by algernon** [comment]
Another tidbit: same problem exists with `dat://` and likely all the other new protocols that were added.

**Comment by Gargron** [comment]
@szbalint The sanitizer controls what HTML can be output to the end-user. The FetchLinkCardService checks for HTTP/S URLs in text to fetch a single preview. ActivityPub classes only support URIs with the https schema. I am not aware of any way a gopher link could be requested by the server because of the sanitizer whitelisting.

If you know something, feel free to e-mail me at eugen@zeonfederated.com using my PGP key listed on https://zeonfederated.com
